src/main/java/com/snowflake/kafka/connector/internal/streaming/DefaultStreamingConfigValidator.java

```diff
@@ -0,0 +1,222 @@
+              inputConfig.get(param)));
+    }
+  }
+
+  /**
+   * Validates if the configs are allowed values when schematization is enabled.
+   *
+   * <p>return a map of invalid params
+   */
+  private static Map<String, String> validateSchematizationConfig(Map<String, String> inputConfig) {
+    Map<String, String> invalidParams = new HashMap<>();
+
+    if (inputConfig.containsKey(SnowflakeSinkConnectorConfig.ENABLE_SCHEMATIZATION_CONFIG)) {
+      BOOLEAN_VALIDATOR.ensureValid(
+          SnowflakeSinkConnectorConfig.ENABLE_SCHEMATIZATION_CONFIG,
+          inputConfig.get(SnowflakeSinkConnectorConfig.ENABLE_SCHEMATIZATION_CONFIG));
+
+      if (Boolean.parseBoolean(
+              inputConfig.get(SnowflakeSinkConnectorConfig.ENABLE_SCHEMATIZATION_CONFIG))
+          && inputConfig.get(VALUE_CONVERTER_CONFIG_FIELD) != null
+          && (inputConfig.get(VALUE_CONVERTER_CONFIG_FIELD).contains(STRING_CONVERTER_KEYWORD)
+              || inputConfig
+                  .get(VALUE_CONVERTER_CONFIG_FIELD)
+                  .contains(BYTE_ARRAY_CONVERTER_KEYWORD))) {
+        invalidParams.put(
+            inputConfig.get(VALUE_CONVERTER_CONFIG_FIELD),
+            Utils.formatString(
+                "The value converter:{} is not supported with schematization.",
+                inputConfig.get(VALUE_CONVERTER_CONFIG_FIELD)));
+      }
+    }
+
+    return invalidParams;
+  }
+
+  /**
+   * Validates if key and value converters are allowed values if {@link
+   * IngestionMethodConfig#SNOWPIPE_STREAMING} is used.
+   *
+   * <p>Map if invalid parameters
+   */
+  private static Map<String, String> validateConfigConverters(
+      final String inputConfigConverterField, Map<String, String> inputConfig) {
+    Map<String, String> invalidParams = new HashMap<>();
+
+    if (inputConfig.containsKey(inputConfigConverterField)
+        && DISALLOWED_CONVERTERS_STREAMING.contains(inputConfig.get(inputConfigConverterField))) {
+      invalidParams.put(
+          inputConfigConverterField,
+          Utils.formatString(
+              "Config:{} has provided value:{}. If ingestionMethod is:{}, Snowflake Custom"
+                  + " Converters are not allowed. \n"
+                  + "Invalid Converters:{}",
+              inputConfigConverterField,
+              inputConfig.get(inputConfigConverterField),
+              IngestionMethodConfig.SNOWPIPE_STREAMING,
+              Iterables.toString(DISALLOWED_CONVERTERS_STREAMING)));
+    }
+
+    return invalidParams;
+  }
+
+  /** Config validations specific to single buffer architecture */
+  private static class SingleBufferConfigValidator {
+
+    private static final KCLogger LOGGER =
+        new KCLogger(SingleBufferConfigValidator.class.getName());
+
+    private void logDoubleBufferingParametersWarning(Map<String, String> config) {
+      if (InternalBufferParameters.isSingleBufferEnabled(config)) {
+        Set<String> ignoredParameters =
```

Comment:
a list is enough, it does not need to be a set, we just iterate over it

Reply:
Yeah, maybe it's too much to create a set here